trial-config-service/src/main/java/uk/ac/ox/ndph/mts/trial_config_service/controller/TrialConfigController.java

```diff
@@ -0,0 +1,106 @@
+package uk.ac.ox.ndph.mts.trial_config_service.controller;
+
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.reactive.function.client.WebClientException;
+import reactor.core.publisher.Mono;
+import uk.ac.ox.ndph.mts.trial_config_service.config.GitRepo;
+import uk.ac.ox.ndph.mts.trial_config_service.exception.InvalidConfigException;
+import uk.ac.ox.ndph.mts.trial_config_service.exception.ResourceAlreadyExistsException;
+import uk.ac.ox.ndph.mts.trial_config_service.model.Trial;
+import uk.ac.ox.ndph.mts.trial_config_service.service.TrialConfigService;
+import java.io.IOException;
+
+
+
+@RestController
+@RequestMapping("/trial-config")
+public class TrialConfigController {
+
+    private final TrialConfigService trialConfigService;
+    private final WebClient webClient;
+    private GitRepo gitRepo;
+
+    //TODO(darrensmithson) - only temp until we decide where to get userId from
+    String userId = "admin001";
+
+    @Autowired
+    public TrialConfigController(TrialConfigService trialConfigService, WebClient webClient, GitRepo gitRepo) {
+        this.trialConfigService = trialConfigService;
+        this.webClient = webClient;
+        this.gitRepo = gitRepo;
+    }
+
+    public TrialConfigController(TrialConfigService trialConfigService, String baseUrl) {
+        this.trialConfigService = trialConfigService;
+        this.webClient = WebClient.create(baseUrl);
+    }
+
+    @GetMapping("/trial")
```

Comment:
why is GET /trial
creates a repo

and POST /trial
also creates a repo (from json)

Seems like the first should also be a post with empty body. wdyt

Reply:
Yeah I understand the confusion - I created a POST to accept the JSON data in the requestBody - this is correct as we create a Trial (and child objects) from the Json data. The other endpoint 'GET' is used to read data from a file retrieved from GIT, this also creates a Trial (and child objects). The reason for having the slightly incorrect use of GET is this - 
I wanted to keep the endpoints - trail/trial-config the same for both methods of reading/creating Trials and only have the verb as the difference. Once we get close to finalising the solution and decide whether to use GIT or JSON fed directly into the request we can remove 1 method and ensure that it is a POST. Having the endpoints the same but using different (even tho its slightly incorrect) seemed the best option as the changes to test scripts etc would smaller when changing the verb rather than changing the endpoint when we decide to use only 1 way to create a trial. The other option of changing the endpoint (to ensure we dont get an ambiguous mapping exception) per data retrieval type would also be ugly IMO - e.g trial-config/trial/git, trial-config/trial/json..

Reply:
ok. so lets just add a comment that this should be later changed

Reply:
added a TODO in the code as per Elads suggestion